fecgov/openFEC
# Add all-day cache between 9am and 7:30pm ET

## Summary (required)

- Resolves #3545: Add caching between 9am and 7:30pm ET

## How to test the changes locally
 This is easiest to test with a manual deploy, because cloud.gov apps are in UTC and our local environments are on local time.
Only `stage` and `prod` have the API umbrella, so on `dev` all we can check is that the `cache-control` headers are returning the proper `max-age` value. I tested this with a manual deploy to `stage`.
- Between 9am and 7:30pm ET, `curl -I -X GET "https://fec-dev-api.app.cloud.gov/v1/schedules/schedule_a/"` should return `Expires: <today's date, ie. Mon, 18 Mar 2019> 23:00:00 GMT` (7:30pm ET)
- Outside of those times (after 7:30pm until 9am the next day), `curl -I -X GET "https://fec-dev-api.app.cloud.gov/v1/schedules/schedule_a/"` should return `cache-control: public, max-age=3600`

Test response headers:
```
curl -X GET -I "https://api-stage.open.fec.gov/v1/schedules/schedule_a/?api_key=DEMO_KEY"

curl -X GET -I  "https://api-stage.open.fec.gov/v1/schedules/schedule_a/?api_key=DEMO_KEY&contributor_name=smith"

curl -X GET -I  "https://api-stage.open.fec.gov/v1/schedules/schedule_a/?api_key=DEMO_KEY&committee_id=C00213512"

curl -X GET -I   "https://api-stage.open.fec.gov/v1/schedules/schedule_a/?api_key=DEMO_KEY&recipient_committee_type=P"
```

Test query times:
```
 curl -s -w '\nLookup time:\t%{time_namelookup}\nConnect time:\t%{time_connect}\nPreXfer time:\t%{time_pretransfer}\nStartXfer time:\t%{time_starttransfer}\n\nTotal time:\t%{time_total}\n' -o /dev/null "https://api-stage.open.fec.gov/v1/schedules/schedule_a/?api_key=DEMO_KEY"

 curl -s -w '\nLookup time:\t%{time_namelookup}\nConnect time:\t%{time_connect}\nPreXfer time:\t%{time_pretransfer}\nStartXfer time:\t%{time_starttransfer}\n\nTotal time:\t%{time_total}\n' -o /dev/null  "https://api-stage.open.fec.gov/v1/schedules/schedule_a/?api_key=DEMO_KEY&contributor_name=smith"

 curl -s -w '\nLookup time:\t%{time_namelookup}\nConnect time:\t%{time_connect}\nPreXfer time:\t%{time_pretransfer}\nStartXfer time:\t%{time_starttransfer}\n\nTotal time:\t%{time_total}\n' -o /dev/null  "https://api-stage.open.fec.gov/v1/schedules/schedule_a/?api_key=DEMO_KEY&committee_id=C00213512"

 curl -s -w '\nLookup time:\t%{time_namelookup}\nConnect time:\t%{time_connect}\nPreXfer time:\t%{time_pretransfer}\nStartXfer time:\t%{time_starttransfer}\n\nTotal time:\t%{time_total}\n' -o /dev/null  "https://api-stage.open.fec.gov/v1/schedules/schedule_a/?api_key=DEMO_KEY&recipient_committee_type=P"
```
## Impacted areas of the application
List general components of the application that this PR will affect:

-  All endpoints besides /efile/, /legal/, and /calendar-dates/
